test/vars/MavenSpec.groovy

```diff
@@ -7,6 +7,54 @@ class MavenSpec extends JenkinsPipelineSpecification {
         mavenGroovy = loadPipelineScriptForTest("vars/maven.groovy")
     }
 
+    def "[maven.groovy] run Maven"() {
+        when:
+        mavenGroovy.runMaven("clean install")
+        then:
+        1 * getPipelineMock("sh")('mvn -B clean install')
+    }
+
+    def "[maven.groovy] run Maven with option"() {
+        when:
+        mavenGroovy.runMaven("clean install", ['-fae'])
+        then:
+        1 * getPipelineMock("sh")('mvn -B -fae clean install')
+    }
+
+    def "[maven.groovy] run Maven with log file"() {
+        setup:
+        Properties props = new Properties()
+        props.setProperty("anykey", "anyvalue")
+        when:
+        mavenGroovy.runMaven("clean install", ['-fae'], props, "logFile.txt")
+        then:
+        1 * getPipelineMock("sh")('mvn -B -fae clean install -Danykey=anyvalue | tee $WORKSPACE/logFile.txt ; test ${PIPESTATUS[0]} -eq 0')
+    }
+
+    def "[maven.groovy] run Maven with skip tests"() {
+        when:
+        mavenGroovy.runMaven("clean install", true, ['-fae'], "logFile.txt")
+        then:
+        1 * getPipelineMock("sh")('mvn -B -fae clean install -DskipTests=true | tee $WORKSPACE/logFile.txt ; test ${PIPESTATUS[0]} -eq 0')
```

Comment:
isn't better to use maven option to use `-l build.log` ?

Reply:
I have just taken what was initially in the `runMavenWithSettings` method. Not sure if there was a reason to be implemented like that ...

Reply:
I'd rather not change that in this PR as this could have some impact ...

Reply:
sure, just it hit my eyes ;)